api/coreservice.go

```diff
@@ -243,13 +246,21 @@ func WithAPIStats(stats *nodestats.APILocalStats) Option {
 	}
 }
 
+// WithArchiveSupport is the option to enable archive support
+func WithArchiveSupport(enabled bool) Option {
```

Comment:
The `enable` parameter should be redundant; this option is supposed to enable 'archive mode'.

Reply:
updated